src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/EventRecordConstructor.java

```diff
@@ -163,6 +185,97 @@ private GenericRecord getPostHookEvent(HookContext hookContext, EventStatus stat
     return recordBuilder.build();
   }
 
+  private void retrieveApplicationResourceUsageReport(
+      GenericRecordBuilder recordBuilder,
+      ApplicationResourceUsageReport applicationResourceUsageReport) {
+    String reportClassName = "org.apache.hadoop.yarn.api.records.ApplicationResourceUsageReport";
+    String resourceClassName = "org.apache.hadoop.yarn.api.records.Resource";
+
+    ImmutableMap<String, String> reportMethods =
+        ImmutableMap.<String, String>builder()
+            .put("getNumUsedContainers", "YarnReportNumUsedContainers")
```

Comment:
super nit: (do weigh the pro-s and con-s yourself first) The names seem to have a lot in common. Can we do something like - 
```
Stream.of("NumUsedContainers", "NumReservedContainers"...).forEach(name -> callMethodWithReflection(
                "get" + name,
                "YarnReport" + name,
                recordBuilder,
                applicationResourceUsageReport,
                reportClassName))
```